From 6ec4b8e240c277239399d8057551c731a66d3e9a Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 8 Sep 2014 13:06:39 -0400
Subject: [tools] perf/mem: Introduce --list-cmds for use by scripts

Message-id: <1410181752-25631-155-git-send-email-jolsa@redhat.com>
Patchwork-id: 91800
O-Subject: [PATCH RHEL7.1 BZ1134356 154/307] perf mem: Introduce --list-cmds for use by scripts
Bugzilla: 1134356
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1134356
https://bugzilla.redhat.com/show_bug.cgi?id=1134356

upstream
========
commit 8d2a2a1d885d36e978c4619a3db8791f9768dd6a
Author: Ramkumar Ramachandra <artagnon@gmail.com>
Date: Fri Mar 14 23:17:52 2014 -0400
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-mem.c b/tools/perf/builtin-mem.c
index 2e3ade6..4a1a6c9 100644
--- a/tools/perf/builtin-mem.c
+++ b/tools/perf/builtin-mem.c
@@ -21,11 +21,6 @@ struct perf_mem {
  DECLARE_BITMAP(cpu_bitmap, MAX_NR_CPUS);
 };
 
-static const char * const mem_usage[] = {
- "perf mem [<options>] {record <command> |report}",
- NULL
-};
-
 static int __cmd_record(int argc, const char **argv)
 {
  int rec_argc, i = 0, j;
@@ -220,9 +215,15 @@ int cmd_mem(int argc, const char **argv, const char *prefix __maybe_unused)
      " between columns '.' is reserved."),
  OPT_END()
  };
+ const char *const mem_subcommands[] = { "record", "report", NULL };
+ const char *mem_usage[] = {
+  NULL,
+  NULL
+ };
+
 
- argc = parse_options(argc, argv, mem_options, mem_usage,
-        PARSE_OPT_STOP_AT_NON_OPTION);
+ argc = parse_options_subcommand(argc, argv, mem_options, mem_subcommands,
+     mem_usage, PARSE_OPT_STOP_AT_NON_OPTION);
 
  if (!argc || !(strncmp(argv[0], "rec", 3) || mem_operation))
   usage_with_options(mem_usage, mem_options);
diff --git a/tools/perf/perf-completion.sh b/tools/perf/perf-completion.sh
index 0ef59dd..f44c04b 100644
--- a/tools/perf/perf-completion.sh
+++ b/tools/perf/perf-completion.sh
@@ -122,7 +122,7 @@ __perf_main ()
   evts=$($cmd list --raw-dump)
   __perfcomp_colon "$evts" "$cur"
  # List subcommands for perf commands
- elif [[ $prev == @(kvm|kmem) ]]; then
+ elif [[ $prev == @(kvm|kmem|mem) ]]; then
   subcmds=$($cmd $prev --list-cmds)
   __perfcomp_colon "$subcmds" "$cur"
  # List long option names
-- 
1.7.1